triton/tut/array.rst

```diff
@@ -4,249 +4,305 @@ Array jobs
-
-In the example below, we have the same program, but different command
-line parameters. In this case, everything is hard coded in the bash
-script itself. You could also do this directly inside your program, and
-generate the parameters according to some algorithm. This can be really
-powerful: not only can you both hard code and generate with algorithm,
-but if you never edit already-run parameters, you have a single place
-where you can see everything that has been run before. (Personally, I
-always try to set up a system where parameters are defined in one place,
-code in another, and I can always know what has been run for each output
-by looking in just one place. If you are going to be configuring stuff
-by hand anyway, better to have it all together.)::
 
-    #!/bin/bash
-    #SBATCH -n 1
-    #SBATCH -t 04:00:00
-    #SBATCH --mem-per-cpu=2500
-    #SBATCH --array=0-4
-
-    case $SLURM_ARRAY_TASK_ID in
-        0) ARGS="-res 5 -arg 3" ;;
-        1) ARGS="-res 6 -arg 3" ;;
-        2) ARGS="-res 5 -arg 4" ;;
-        3) ARGS="-res 6 -arg 4" ;;
-        4) ARGS="-res 5 -arg 5" ;;
-    esac
-
-    cd $SLURM_ARRAY_TASK_ID
-    srun I am doing $ARGS !
-    # e.g. python input.py $ARGS
-    cd ..
-
-Read parameters from file
-~~~~~~~~~~~~~~~~~~~~~~~~~
-
-Now we do basically the same thing as above, but we have all of the
-parameters stored in another file ``arrayparams.txt``::
-
-  input_561 --opt1
-  input_418 --opt2
-  input_569 --opt1
-
-In our script, we use the ``sed`` program to read just one line from
-the file.  This is stored in the variable ``line``, and then we can
-use this however: in this case by using it as the parameters to the
-program.  Don't worry about how the ``sed`` command works - no one
-really knows, we just find it via a web search.  Note that the line
-numbers start at one, not zero!
+Hardcoding arguments in the batch script
+----------------------------------------
+
+One way to pass arguments to your code is by hardcoding them in the batch script 
+you want to submit to Slurm. 
+
+Assume you would like to run the Pi estimation code for 5 different seed values, each
+for 2.5 million iterations. You could assign a seed value to each task in you job array
+and save each output to a file. Having calculated all estimations, you could take the 
+average of all the Pi values to arrive at a more accurate estimate. An example of such
+a batch script is as follows. 
+
+.. code-block:: bash
+
+   #!/bin/bash 
+   #SBATCH --job-name=pi_estimation
+   #SBATCH --output=/dev/null
+   #SBATCH --error=/dev/null
+   #SBATCH --array=0-4
+   #SBATCH --time=01:00:00
+   #SBATCH --mem=500
+   # You may put the commands after SBATCH directives
+
+   case $SLURM_ARRAY_TASK_ID in
+
+       0)
+           SEED=123
+           ;;
+       1)  
+           SEED=38
+           ;;
+       2)  
+           SEED=22 
+           ;;
+       3) 
+           SEED=60 
+           ;; 
+       4) 
+           SEED=432 
+           ;;
```

Comment:
This is a case where I would definitely leave it in long form, so that it can be most easily edited and you can easily compare the different lines, and keep them in sync and all.